### PR TITLE
[RTM] Accommodate Sphinx >=1.6 warning behavior

### LINKED ISSUE #521: Circle builds failing due to old AFNI
Looks like Sphinx has a [new release out today](https://pypi.python.org/pypi/Sphinx/1.6.1), so we're getting nipype warnings raised to Sphinx warnings, which are treated as errors.

```
AFNI is outdated, detected version Debian-16.2.07~dfsg.1-2~nd16.04+1 and AFNI_17.1.05 is available.
```

### PR DESCRIPTION
**Pull request by effigies**
Closes #521

### CODE REVIEW

**Comment by chrisfilo** [comment]
Suppressing all warnings is a bit dangerous. Could we do something more nuanced using filtering? https://docs.python.org/3.1/library/warnings.html#the-warnings-filter

**Comment by effigies** [comment]
I'll look into it.

**Comment by effigies** [comment]
Logger warnings are homogeneous, as far as I can see. I can switch to reducing warnings in `docs/conf.py`, rather than in the program in general.

**Comment by chrisfilo** [comment]
It seem that you can filter warning using regular expression on the text of the warning (unless I am misreading the documentation). This would allow us to suppress warning that we know can be ignored (for example the AFNI version).

**Comment by effigies** [comment]
Gotcha. Pushed a few filters. Also included some that I saw in the docs output, even though they don't seem to be causing sphinx to throw up. We've had users mention the cmp warning a couple times, I think.

Let me know what you think.

**Comment by effigies** [comment]
Well, looks like ignored warnings can still be caught by Sphinx. Want to keep the fitlers, but restore the `setLevel`s from 0815732?

**Comment by chrisfilo** [comment]
Does the filtering of warnings by content work at all? I cannot see since all the CircleCI builds are docs only.

**Comment by effigies** [comment]
Amended the commit so the full build will run.

**Comment by chrisfilo** [comment]
The warnings seem to still be there when running pipelines:

```
170516-21:29:49,255 interface WARNING:

	 AFNI is outdated, detected version Debian-16.2.07~dfsg.1-2~nd16.04+1 and AFNI_17.1.05 is available.
```

**Comment by effigies** [comment]
I don't think we can suppress logged warnings without changing the log level.

However, search for "Fieldmap estimation:" in the output of (container 0) builds [440](https://circleci.com/gh/effigies/fmriprep/440) and [428](https://circleci.com/gh/effigies/fmriprep/428). In 440, the `cmp not installed` warning is suppressed from the output.

**Comment by chrisfilo** [comment]
I see - so such suppression would require changes at nipype level right?

**Comment by effigies** [comment]
I'm not sure what they'd do differently. They use the standard logger to log messages of different [criticality](https://docs.python.org/3/library/logging.html#logging-levels).

If our goal is to suppress the AFNI message, we could monkey-patch `interfaces.afni.Info` to not check for new AFNI. This might be a good idea anyway, so that running FMRIPREP doesn't have to try to make an outgoing HTTP connection every time an AFNI interface that checks version information is run.

Or we could try to convince Neurodebian to upgrade their AFNI packages.

**Comment by chrisfilo** [comment]
Would it be hard to monkey patch this or should we do it in another PR?

**Comment by effigies** [comment]
Proposed a monkey patch. We'll see how it goes.

**Comment by effigies** [comment]
Monkey patch seems to work. matplotlib warnings aren't suppressed in doc building, and only show up in doc building, so not sure those are worth filtering.

cmp warning does show up in container 0 runtime tests, and this does filter those. So I'll remove the AFNI and matplotlib filters, but keep cmp. Sound reasonable?

File: docs/conf.py [hunk 1]
```diff
@@ -15,6 +15,11 @@
 import os
 import sys
 import subprocess
+from nipype import logging
+
+# Disable informational interface warnings
+logging.getLogger('interface').setLevel('ERROR')
+logging.getLogger('workflow').setLevel('ERROR')
```
Comment (chrisfilo) [inline docs/conf.py]: If monkey patching works this should not be necessary anymore right?

Comment (effigies) [inline docs/conf.py]: True. Let me drop those. Do we need a changelog entry?

**Comment by chrisfilo** [comment]
I think we can skip it for this one.

On Wed, May 17, 2017 at 10:25 AM, Chris Markiewicz <notifications@github.com

**Comment by effigies** [comment]
Oh right. We're getting [this warning](https://github.com/poldracklab/fmriprep/blob/master/fmriprep/workflows/epi.py#L57) now. That at least was being suppressed by setting the workflow logger level to 'ERROR' in `docs/conf.py`. Should I revert to that, or do you have another preferred solution?

**Comment by chrisfilo** [comment]
I would prefer not to do a blanket warning suppression. Changing the level of this one log message to INFO seems less error prone.

**Pull request merged**